docs/src/convert-to-onda.md

```diff
@@ -0,0 +1,264 @@
+
+The main factor to consider when planning conversion of a large batch of EDF files is that planning requires only the (small number) of header bytes, even for very large EDF files.
+Thus, the first step is to read the file headers into memory without reading the signal data itself (which for more than a few EDF files will not in general fit into memory ).
+
+#### Reading headers from local filesystem
+
+For EDF files stored on a normal filesystem, the `EDF.File` constructor will by default create a "header-only" `EDF.File`, so multiple files' headers can be read like
+
+```julia
+files = map(edf_paths) do path
+    open(EDF.File, path, "r")
+end
+```
+
+#### Reading headers from S3
+
+!!! note
+    This section may become obsolete in a future version of EDF.jl which uses the [conditional dependency](https://pkgdocs.julialang.org/v1/creating-packages/#Weak-dependencies) functionality available from Julia 1.9+ to provide tighter integration with AWSS3.jl.
+
+Unfortunately, `open(path::S3Path)` will fetch the entire contents of the object stored at `path`, so we need to be a bit clever to read _only_ header bytes from an S3 file, especially given that the number of bytes we need to read depends on the number of signals.
+The following is an example of one technique for reading EDF file and signal headers from S3:
+
+```julia
+function EDF.read_file_header(path::S3Path)
+    bytes = s3_get(path.bucket, path.key; byte_range=1:256)
+    buffer = IOBuffer(bytes)
+    return EDF.read_file_header(buffer)
+end
+
+function EDF.File(path::S3Path)
+    _, n_signals = EDF.read_file_header(path)
+    bytes = s3_get(path.bucket, path.key; byte_range=1:(256 * (n_signals + 1)))
+    return EDF.File(IOBuffer(bytes))
+end
+
+# use asyncmap because this is mostly bound by request roundtrip latency
+files = asyncmap(EDF.File, edf_paths)
+```
+
+#### Concatenating plans into one big table
+
+When doing bulk review of plans, it's generally helpful to have the individual files' plans concatenated into a single large table.
+It's important to keep track of which plan rows corresopnd to which input file, which can be accomplished via something like this:
+
+```julia
+# create a UUID namespace to make recording ID generation idempotent
+const NAMESPACE = UUID(...)
+function plan_all(edf_paths, files; kwargs...)
+    plans = mapreduce(vcat, edf_paths, files) do origin_uri, edf
+        plan = plan_edf_to_onda_samples(edf; kwargs...)
+        plan = DataFrame(plan)
+        # make sure this is the same every time this function is re-run!
+        recording = uuid5(NAMESPACE, string(origin_uri))
+        return insertcols!(plan, 
+                           :origin_uri => origin_uri,
+                           :recording => recording)
+    end
+end
+```
+
+### Review and revise the plans
+
+This "bulk plan" table can then be reviewed in bulk, looking for patterns in which `label`s are not matched, physical units associated with each `sensor_type`, etc.
```

Comment:
Do we want to mention that determining custom labels is sometimes context dependent on what other signals are present in the EDF?

Might already be obvious, not sure how helpful it is to someone. I want thinking this specifically the context of bulk review of multiple EDFs. Sometimes the same EDF label can be interpreted as different things pending on the context and presence of other signal labels in the same EDF (I unfortunately don't have a specific example of this off the top of my head)

Reply:
yeah that's a very good point, I'll add a bit about that (and more caveats about how labels are matched), maybe even in the earlier section ...